.github/workflows/go.yaml

```diff
@@ -0,0 +1,91 @@
+name: Go
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+
+  pull_request:
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+
+permissions:
+  contents: read
+
+jobs:
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+          cache: false
+      - name: fmt
+        run: go fmt ./...
+      - name: Check diff
+        run: git diff --exit-code
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+          cache: false
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          only-new-issues: true
+          version: v1.51.1
+
+  generate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+          cache: false
+      - name: Generate
+        run: make generate
+      - name: Check diff
+        run: git diff --exit-code
+
+  vulncheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+          cache: false
+      - name: Install govulncheck
+        run: go install golang.org/x/vuln/cmd/govulncheck@latest
+      - name: Check vulnerabilities
+        run: govulncheck ./...
```

Comment:
Why do we need a vulnerability check if we're using GitHub Advanced Security? Does this run against a database we don't support?

Reply:
```
Using go1.20.1 and govulncheck@v0.0.0 with
vulnerability data from https://vuln.go.dev/ (last modified 13 Mar 23 20:29 UTC).

Scanning your code and 637 packages across 76 dependent modules for known vulnerabilities...
Your code is affected by 1 vulnerability from the Go standard library.

Vulnerability #1: GO-2023-1621
  The ScalarMult and ScalarBaseMult methods of the P256 Curve may
  return an incorrect result if called with some specific
  unreduced scalars (a scalar larger than the order of the curve).
  This does not impact usages of crypto/ecdsa or crypto/ecdh.

  More info: https://pkg.go.dev/vuln/GO-2023-1621

  Standard library
    Found in: crypto/internal/nistec@go1.20.1
    Fixed in: crypto/internal/nistec@go1.20.2

    Call stacks in your code:
      github/actions/testserver/server.go:94:50: github.com/actions/actions-runner-controller/github/actions/testserver.DefaultActionsToken calls github.com/golang-jwt/jwt/v4.ParseRSAPrivateKeyFromPEM, which eventually calls crypto/internal/nistec.P256Point.ScalarBaseMult
      pkg/githubwebhookdeliveryforwarder/githubwebhookdelivery.go:201:13: github.com/actions/actions-runner-controller/pkg/githubwebhookdeliveryforwarder.server.Errorf calls fmt.Fprintf, which eventually calls crypto/internal/nistec.P256OrdInverse
      pkg/githubwebhookdeliveryforwarder/githubwebhookdelivery.go:201:13: github.com/actions/actions-runner-controller/pkg/githubwebhookdeliveryforwarder.server.Errorf calls fmt.Fprintf, which eventually calls crypto/internal/nistec.P256Point.ScalarMult
```